runtime/server/batch_query.go

```diff
@@ -0,0 +1,177 @@
+package server
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	runtimev1 "github.com/rilldata/rill/proto/gen/rill/runtime/v1"
+	"github.com/rilldata/rill/runtime/server/auth"
+)
+
+func (s *Server) QueryBatch(req *runtimev1.QueryBatchRequest, srv runtimev1.QueryService_QueryBatchServer) error {
+	ctx := srv.Context()
+
+	if !auth.GetClaims(ctx).CanInstance(req.InstanceId, auth.ReadOLAP) {
```

Comment:
`ReadOLAP` actually gives access to run any SQL query, so it's not available in cloud. We would need to check for `ReadMetrics` (for metrics view APIs) and `ReadProfiling` (for modeler APIs) individually for each query. (I guess this already happens?)

Reply:
Yes. Have added a TODO to check for access upfront.